### DISCUSSION TITLE
How to connect to a TCP/IP port when using Feildtrip Buffer

### DISCUSSION THREAD

**Discussion by Fang-UM** [umair-hassan/best-toolbox#35] [General]
Hi. I am using BEST Toolbox to establish a closed-loop tACS-EEG protocol. However, the BEST Toolbox couldn't received my EEG signal transferred from BrainVision Recorder RDA via a outside Fieldtrip Buffer built by myselves. Could you please indicate how should I transfer rsEEG data from a specific Fieldtrip Buffer?

**Comment by umair-hassan** [comment]
Dear Fang,

Thanks for your message. Can you elaborate a bit more on how do you plan to use the EEG for closing the loop for tACS using the BEST Toolbox? I am assuming that you plan to use the TTL triggers to deliver a predefined protocol on the tACS side ? On EEG side, the BEST Toolbox offers, ERP Measurement protocol and TEP measurements as well as brain state dependent (real-time approach). Please note that the brain-state dependent protocol are dependent on sync2brain bossdevice for intended closed-loop functionality at the moment, but we are open to work on an open project to make it work otherwise with open source buffers.

For BVR RDA, have you downloaded Peter Rydesäter's (2021). TCP/UDP/IP Toolbox 2.0.6 from Matlab? If so, then you will have to double check whether difference exist in pnet.m of RDA client and this toolbox.

**Comment by Fang-UM** [comment]
Dear Umair
Thank you for your quick reply! About how do we establish the closed loop protocol, we plan to use the BEST Toolbox to collect rsEEG data from BVR RDA, while sending TTL triggers to trigger an external tACS controllor depending on the real-time rsEEG data. However, our devises are not sync2brain boss device. In this case, these two routins could not work at the same time, right? Thank you again for your help!

**Comment by umair-hassan** [comment]
Hi Fang,

Yes you are right, currently these two requirements (real-time rs-EEG power triggered protocol capability of best toolbox & BVR RDA) cannot work together. However, I am working on a new release that will make the BVR RDA directly compatible with the best toolbox without being streamed from real-time buffers of fieldtrip/BCI2000, although this will enable non real-time triggered protocol capability already possible, but I will need to work a bit more on to take it to real-time rsEEG compatible. I will post the best toolbox release link over here, I am also glad to have a meeting with you on your setup in order to set up these things. Let me know your preferences. If you want to join user meeting already tomorrow, I will be happy to provide the zoom link, it is going to happen at 0930 German time. 

Thanks, Umair

**Comment by Fang-UM** [comment]
Hi Umair,

I'm so glad to be invited to your meeting! It is available for me and my colleague to join in at that time! Thank you!

Sincerely,

Fang

________________________________

**Comment by umair-hassan** [comment]
Hi Fang,

This notification didn't reach me on time unfortunately, should we schedule a call together at some point in coming days to dicuss and implement your setup?

Thanks, Umair

**Comment by Fang-UM** [comment]
Hi Umair,
Good to hear from you! I’m sorry for the late reply about my last email.
Are you free tomorrow 13-15h (German time)? Or do you have other time preferences? Thank you!
Sincerely,
Fang

**Comment by umair-hassan** [comment]
Hi Fang,

Tomorrow, I will be a little bit busy but Friday evening 5pm onwards German time is a good option, do you think its possible? Then we can set some zoom link.

Thanks, Umair

**Comment by Fang-UM** [comment]
Hi Umair

I am sorry that Friday 5pm is not available for me. I wonder is Monday 15h onwards available for you? Thank you!

Bests,

Fang

**Comment by umair-hassan** [comment]
Hi Fang, 

We can meet monday at 18h onwards if thats fine with you? 

I will send the zoom then.

Thanks Umair

**Comment by Fang-UM** [comment]
?Hi Umair

That's fine! Thank you. See you then.

Sincerely,

Zhou

________________________________

**Comment by Fang-UM** [comment]
Hi Umair,
I wonder If there is a zoom link here? Thank you!
Bests,
Zhou

**Comment by umair-hassan** [comment]
Hi Zhou, I will give you a link shortly here

**Comment by umair-hassan** [comment]
@Fang-UM https://us04web.zoom.us/j/72883818024?pwd=SVdVRjRTU21nWUVPSElpYzJBS3dEUT09
